### ISSUE TITLE
Union type as second arg for `isinstance` raises W1116

### ISSUE THREAD

**Issue by hofrob** [PyCQA/pylint#8381]
### Bug description

The following is valid code in python 3.10+ (iirc):

```py
a = 1

isinstance(a, int | str)
```

### Configuration

```ini
# default
```

### Command used

```shell
pylint --score=n -d all -e W1116 foo.py
```

### Pylint output

```shell
************* Module foo
foo.py:3:0: W1116: Second argument of isinstance is not a type (isinstance-second-argument-not-valid-type)
```

### Expected behavior

No error

### Pylint version

```shell
$ pylint --version
pylint 2.16.3
astroid 2.14.2
Python 3.11.2 (main, Feb  9 2023, 13:29:03) [GCC 11.3.0]
```

### OS / Environment

Ubuntu 22.04

**Comment by hofrob** [comment]
I just discovered this because ruff is rewriting `isinstance(a, (int, str))` to union types since v0.0.254. It may be related to another issue, but I came up empty after looking at existing ones.

**Comment by nickdrozd** [comment]
Fixed with https://github.com/PyCQA/pylint/pull/8206. 2.17 maybe?

**Comment by Pierre-Sassoulas** [comment]
We'll release 2.17 soon, we're only waiting for try star node support from astroid.
